x-pack/test/stack_functional_integration/test/functional/apps/reporting/util.js

```diff
@@ -0,0 +1,44 @@
+import expect from '@kbn/expect';
+
+export const dash = x => x.replace(/\s/gm, '-');
+export const pretty = x =>
+  JSON.stringify(x, null, 2);
+export const buildUrl = ({ protocol, auth, hostname, port }) =>
+  new URL(`${protocol}://${auth}@${hostname}:${port}`);
+export const putWatcher = async (watch, id, body, client, log) => {
+  const putWatchResponse = await client.watcher.putWatch({ ...watch, body });
+  log.debug(`# putWatchResponse \n${pretty(putWatchResponse)}`);
+  expect(putWatchResponse.body._id).to.eql(id);
+  expect(putWatchResponse.statusCode).to.eql('201');
+  expect(putWatchResponse.body._version).to.eql('1');
+};
+export const getWatcher = async (watch, id, client, log, common, tryForTime) => {
+  await common.sleep(20000);
+  await tryForTime(90000, async () => {
+    await common.sleep(15000);
+
+    // *** Prolly gonna drop the following commented out code soon!
+    // History search
+    // const watcherHistoryResponse = await esClient.watcher.getWatch(watchId);
+    // Print history search result
+    // log.debug(`\nwatcherHistoryResponse=\n${pretty(watcherHistoryResponse)}\n`);
+
```

Comment:
I think we need to get the commented-out code working.  I just had a failure and you really can't see the root cause without seeing that watcher-history data.  For example here's the console output for this failing test;
```
       └-> should be successful and increment revision
         └-> "before each" hook: global before each
         │ debg ... sleep(20000) start
         │ debg ... sleep(20000) end
         │ debg ... sleep(15000) start
         │ debg ... sleep(15000) end
         │ debg --- retry.tryForTime error: expected 1 to be above 1
         │ debg
         │      ### tryForTime-Failure--raw body:
         │              undefined
         │ debg ... sleep(15000) start
         │ debg ... sleep(15000) end
         │ debg --- retry.tryForTime failed again with the same message...
         │ debg
         │      ### tryForTime-Failure--raw body:
         │              undefined
         │ debg ... sleep(15000) start
         │ debg ... sleep(15000) end
         │ debg
         │       getWatchResponse.body._version: 2
         │ debg --- retry.tryForTime error: expected false to sort of equal true
         │ debg
         │      ### tryForTime-Failure--raw body:
         │              undefined
         │ debg ... sleep(15000) start
         │ debg ... sleep(15000) end
         │ debg
         │       getWatchResponse.body._version: 2
         │ debg --- retry.tryForTime failed again with the same message...
         │ debg
         │      ### tryForTime-Failure--raw body:
         │              undefined
         │ debg ... sleep(15000) start
         │ debg ... sleep(15000) end
         │ debg
         │       getWatchResponse.body._version: 2
         │ debg --- retry.tryForTime failed again with the same message...
         │ debg
         │      ### tryForTime-Failure--raw body:
         │              undefined
         │ debg ... sleep(15000) start
         │ debg ... sleep(15000) end
         │ debg
         │       getWatchResponse.body._version: 2
         │ debg --- retry.tryForTime failed again with the same message...
         │ info Taking screenshot "C:\Users\LeeDr\git\kibana\x-pack\test\functional\screenshots\failure\reporting app watcher app PNG Reporting watch should be successful and increment revision.png"
         │ info Current URL is: http://localhost:5601/app/monitoring#/no-data?_g=h@44136fa
         │ info Saving page source to: C:\Users\LeeDr\git\kibana\x-pack\test\functional\failure_debug\html\reporting app watcher app PNG Reporting watch should be successful and increment revision.html

  1) reporting app
       watcher app
         PNG Reporting watch
           should be successful and increment revision:
     Error: retry.tryForTime timeout: Error: expected false to sort of equal true
    at Assertion.assert (C:\Users\LeeDr\git\kibana\packages\kbn-expect\expect.js:100:11)
    at Assertion.eql (C:\Users\LeeDr\git\kibana\packages\kbn-expect\expect.js:235:8)
    at eql (C:\Users\LeeDr\git\kibana\x-pack\test\stack_functional_integration\test\functional\apps\reporting/util.js:31:91)
    at process._tickCallback (internal/process/next_tick.js:68:7)
      at lastError (C:\Users\LeeDr\git\kibana\test\common\services\retry/retry_for_success.ts:28:9)
      at onFailure (C:\Users\LeeDr\git\kibana\test\common\services\retry/retry_for_success.ts:68:13)

         └- × fail: "reporting app watcher app PNG Reporting watch should be successful and increment revision"
```

But if I go into Kibana and add an index pattern for .watcher-history I can see why it failed;
```
                "type": "messaging_exception",
                "reason": "failed to send email with subject [PNG 8.0.0 watcher_png_report-Today-at-12:32-PM, VM=ubuntu16_tar d5cbb474ebc2daa4075911b62e6a2685434f63ed by:LeeDr-XPS] via account [gmail_account]"
              }
            ],
            "type": "messaging_exception",
            "reason": "failed to send email with subject [PNG 8.0.0 watcher_png_report-Today-at-12:32-PM, VM=ubuntu16_tar d5cbb474ebc2daa4075911b62e6a2685434f63ed by:LeeDr-XPS] via account [gmail_account]",
            "caused_by": {
              "type": "messaging_exception",
              "reason": "Could not convert socket to TLS",
              "caused_by": {
                "type": "s_s_l_handshake_exception",
                "reason": "PKIX path building failed: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target",
                "caused_by": {
                  "type": "validator_exception",
                  "reason": "PKIX path building failed: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target",
                  "caused_by": {
                    "type": "sun_cert_path_builder_exception",
                    "reason": "unable to find valid certification path to requested target"
                  }
```

Reply:
Oh that's a great point!  Thanks sir!